File: attendancemanagement.client/src/consts/types/constBase.ts
/** ベース定義クラス */
export default class ConstBase {

    /** ID */
    private _id: string;

    /**
     *  コンストラクタ
     * @param id ID
     */
    protected constructor(id: string) {
        this._id = id;
    }

    /// <summary>
    /// ID
    /// </summary>
    public get id(): string {
        return this._id;
    }
}

File: Common/Const/class/ConstBase.cs
namespace Common.Const
{
    /// <summary>
    /// 定義ベースクラス
    /// </summary>
    public class ConstBase
    {
        /// <summary>
        /// コンストラクタ
        /// </summary>
        /// <param name="id">ID</param>
        protected ConstBase(string id)
        {
            this.Id = id;
        }

        /// <summary>
        /// ID
        /// </summary>
        public string Id { get; }
    }
}
